src/roles/crazed_shaman.py

```diff
@@ -0,0 +1,180 @@
+import re
+import random
+import itertools
+from collections import defaultdict, deque
+
+import botconfig
+from src.utilities import *
+from src import debuglog, errlog, plog, users, channels
+from src.functions import get_players, get_all_players, get_main_role, get_reveal_role, get_target
+from src.decorators import command, event_listener
+from src.containers import UserList, UserSet, UserDict, DefaultUserDict
+from src.dispatcher import MessageDispatcher
+from src.messages import messages
+from src.events import Event
+
+from src.roles._shaman_helper import havetotem, _get_target, _give_totem, _apply_totem
+
+TOTEMS = UserDict()         # type: Dict[users.User, str]
+LASTGIVEN = UserDict()      # type: Dict[users.User, users.User]
+SHAMANS = UserDict()        # type: Dict[users.User, List[users.User]]
+
+@command("give", "totem", chan=False, pm=True, playing=True, silenced=True, phases=("night",), roles=("crazed shaman",))
+def crazed_shaman_totem(var, wrapper, message, prefix="You"):
+    """Give a random totem to a player."""
+
+    target = _get_target(var, wrapper, message, LASTGIVEN)
+    if not target:
+        return
+
+    _give_totem(var, wrapper, target, prefix, set(), "crazed shaman", "", SHAMANS)
```

Comment:
I have a couple of issues with the aesthetics of this call.
1. I don't like how it mutates SHAMANS. Can we have give_totem return the thing we use to add to SHAMANS instead? e.g. SHAMANS[wrapper.source] = give_totem(var, wrapper, target, ...)
2. I don't like passing set() and "" as positional arguments -- it's very non-obvious what those are for. If it's necessary to pass them at all rather than them just taking default values from give_totem, can we pass them as kwargs?